### PR TITLE
koha-suomi

### PR DESCRIPTION
**Pull request by natlibfi-jonollil**
Record-import-transformer styled validator & tests (Melkehitys-1454)
Melinda-oai-pmh-provider styled record formatting  (Melkehitys-1455)

Tested by tests & locally

### CODE REVIEW

**Comment by natlibfi-jonollil** [comment]
Unauthorized: access to the requested resource is not authorized

**Comment by natlibfi-arlehiko** [comment]
Restarted

**Comment by natlibfi-jonollil** [comment]
Tryed to test locally. Gave back 401 duno why

File: src/routes/bib.js [hunk 1]
```diff
@@ -29,107 +29,114 @@
...
-			next(err);
-		}
-	}
+  const CONTENT_TYPES = {
+    'application/json': FORMATS.JSON,
+    'application/marc': FORMATS.ISO2709,
+    'application/xml': FORMATS.MARCXML
+  };
+
+  const ipFilterList = JSON.parse(IP_FILTER_BIB).map(rule => new RegExp(rule)); // eslint-disable-line require-unicode-regexp
```
Comment (natlibfi-arlehiko) [inline src/routes/bib.js]: Don't disable, but fix. Add `u` flag to the RegExp constructor.

File: src/services/validation.js [hunk 2]
```diff
@@ -28,31 +28,22 @@
...
-
-		if (results.valid) {
-			return results.messages;
-		}
-
-		throw new ValidationError(results.messages);
-		*/
-	}
-}
+/* eslint-disable new-cap */
```
Comment (natlibfi-arlehiko) [inline src/services/validation.js]: Don't disable globally. But this shouldn't be needed in this file anyway.

File: src/services/validation.js [hunk 3]
```diff
@@ -28,31 +28,22 @@
...
+import validateFactory from '@natlibfi/marc-record-validate';
+import {
+  FieldExclusion
+} from '@natlibfi/marc-record-validators-melinda';
+
+export default async () => {
+  const validate = validateFactory([await FieldExclusion([{tag: /^003$/u, value: /^(.(?<!FI-MELINDA))*?$/u}])]); // eslint-disable-line prefer-named-capture-group
```
Comment (natlibfi-arlehiko) [inline src/services/validation.js]: Validation should only validate, not fix. If the validation fails the record is considered invalid. See OpenAPI spec for how the error messages are passed to the client. I also updated the Jira task to be more specific about the validation.

Control fields validator is propably the right one for this use case.

File: src/utils.js [hunk 4]
```diff
@@ -31,21 +31,21 @@
...
-	}
+  if (Number.isNaN(Number(value))) {
+    return value === 'true';
+  }
 
-	return Boolean(Number(value));
+  return Boolean(Number(value));
 }
 
 export function createWhitelistMiddleware(whitelist) {
```
Comment (natlibfi-arlehiko) [inline src/utils.js]: As said previously, ditch whitelist middleware. This kind of stuff will be handled in a reverse proxy

File: src/services/validation.spec.js [hunk 5]
```diff
@@ -26,83 +26,66 @@
...
+    '..',
+    'test-fixtures',
+    'validation'
+  ];
+  const {getFixture} = fixtureFactory({root: FIXTURES_PATH});
+  let validator; // eslint-disable-line functional/no-let
+
+  before(async () => {
+    validator = await createValidator();
+  });
```
Comment (natlibfi-arlehiko) [inline src/services/validation.spec.js]: Not necessary. Declare validator as `const` in each unit test

File: package.json [hunk 6]
```diff
@@ -34,57 +34,52 @@
...
-		"sinon": "^7.5.0",
-		"sinon-chai": "^3.3.0"
+		"mocha": "^7.1.0",
+		"nock": "^12.0.2",
+		"npm-check": "^5.9.2",
+		"nyc": "^15.0.0",
+		"sinon": "^9.0.1",
+		"sinon-chai": "^3.5.0"
 	},
 	"eslintConfig": {
```
Comment (natlibfi-arlehiko) [inline package.json]: This still needs to be fixed to:
```
"eslintConfig": {
  "extends": "@natlibfi/melinda-backend"
}
```

**Pull request merged**